src/domain/participants/participants.js

```diff
@@ -414,24 +431,35 @@ const postParticipantsBatch = async (headers, method, requestPayload, span) => {
         clonedHeaders[Enums.Http.Headers.FSPIOP.DESTINATION] = payload.partyList[0].partyId.fspId
         clonedHeaders[Enums.Http.Headers.FSPIOP.SOURCE] = Enums.Http.Headers.FSPIOP.SWITCH.value
       }
-      await participant.sendRequest(clonedHeaders, clonedHeaders[Enums.Http.Headers.FSPIOP.DESTINATION], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT, Enums.Http.RestMethods.PUT, payload, { requestId }, span)
+      await participant.sendRequest(clonedHeaders, clonedHeaders[Enums.Http.Headers.FSPIOP.DESTINATION], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT, Enums.Http.RestMethods.PUT, payload, { requestId }, childSpan)
+      if (childSpan && !childSpan.isFinished) {
```

Comment:
In that particular case, the check for childSpan.isFinished seems like insurance, since the childSpan will be finished in the sendRequest method.
But if the sendRequest is not awaited, then, if someone adopts this practice, the childSpan will be finished before the sendRequest is over.